CMakeLists.txt

```diff
@@ -148,7 +148,7 @@ target_compile_definitions(Common INTERFACE
   GEARSHIFFT_PRECISION_HALF_ONLY=${GEARSHIFFT_PRECISION_HALF_ONLY}
   GEARSHIFFT_PRECISION_SINGLE_ONLY=${GEARSHIFFT_PRECISION_SINGLE_ONLY}
   GEARSHIFFT_PRECISION_DOUBLE_ONLY=${GEARSHIFFT_PRECISION_DOUBLE_ONLY}
-  $<$<BOOL:Boost_USE_STATIC_LIBS>:BOOST_TEST_DYN_LINK>
```

Comment:
when you use standalone tests in gearshifft, which do not have a main method, and when you use boost static libs, then it should give you a linker error like `undefined reference to `main'`. Thus, `BOOST_TEST_DYN_LINK` is given to the files to trigger a generation of a main method by the boost lib. And this changed line is probably the reason for the failed travis tests.

However, I see some fishy defines and includes in benchmark.hpp.
Maybe you can provide the scenario setting, where this method fails.

Reply:
In my humble opinion, I think the logic of the line `$<$<BOOL:Boost_USE_STATIC_LIBS>:BOOST_TEST_DYN_LINK>` is wrong. Its my understanding that you should pass `BOOST_TEST_DYN_LINK` when you do *not* link against static boost. So the way the define is added here seems opposite to my understanding. I build this understanding partially just on the name of the flag: `BOOST_TEST_DYN_LINK` should be defined when *not* `Boost_USE_STATIC_LIBS` is true, otherwise it would be a pretty poorly named flag? :-)